File: bot/services/ai/perplexity_client.js
import fetch from "node-fetch";

export async function sonar(prompt) {
  const res = await fetch("https://api.perplexity.ai/chat/completions", {
    method: "POST",
    headers: {
      "Authorization": `Bearer ${process.env.PERPLEXITY_API_KEY}`,
      "Content-Type": "application/json"
    },
    body: JSON.stringify({
      model: "sonar-pro",
      messages: [{ role: "user", content: prompt }]
    })
  });

  const data = await res.json();
  return data?.choices?.[0]?.message?.content || "No response";
}

File: ai/gcz-sandbox/backend/services/ai/perplexity_client.py
from perplexity import Perplexity
import os

API_KEY = os.environ.get("PERPLEXITY_API_KEY")
client = Perplexity(api_key=API_KEY)

def ask_perplexity(prompt: str, model: str = "sonar-pro") -> str:
    completion = client.chat.completions.create(
        model=model,
        messages=[{"role": "user", "content": prompt}]
    )
    return completion.choices[0].message.content
